packages/kbn-legacy-logging/src/utils/get_payload_size.ts

```diff
@@ -8,12 +8,17 @@
 
 import { isPlainObject } from 'lodash';
 import { ReadStream } from 'fs';
+import { Zlib } from 'zlib';
 import type { ResponseObject } from '@hapi/hapi';
 
 const isBuffer = (obj: unknown): obj is Buffer => Buffer.isBuffer(obj);
 const isFsReadStream = (obj: unknown): obj is ReadStream =>
   typeof obj === 'object' && obj !== null && 'bytesRead' in obj && obj instanceof ReadStream;
+const isZlibStream = (obj: unknown): obj is Zlib => {
+  return typeof obj === 'object' && obj !== null && 'bytesWritten' in obj;
+};
 const isString = (obj: unknown): obj is string => typeof obj === 'string';
+const isObject = (obj: unknown) => isPlainObject(obj) || Array.isArray(obj);
```

Comment:
the name is a bit confusing. let's keep the array check outside:
```js
if(isObject(obj) || Array.isArray(obj)) {
  ...
}
```

We can check that array consists of `simple objects`. But the current implementation is the most practical one.

Reply:
Good point, will pull that out.